lib/Sema/PlaygroundTransform.cpp

```diff
@@ -949,6 +953,24 @@ void swift::performPlaygroundTransform(SourceFile &SF, PlaygroundOptionSet Opts)
     }
 
     PreWalkAction walkToDeclPre(Decl *D) override {
+      // If the selective transform option is enabled then skip any
+      // non-annotated nodes.
+      if (Options.SelectiveTransform) {
+        // Only funcs can be annotated with @_PlaygroundTransformed.
+        auto *funcDecl = dyn_cast<FuncDecl>(D);
+        // In addition to file-level funcs, also support annotated funcs in
+        // structs, classes, extensions and macro expansions.
+        bool isAllowedOuterDecl =
+            (dyn_cast<StructDecl>(D) || dyn_cast<ClassDecl>(D) ||
+             dyn_cast<ExtensionDecl>(D) || dyn_cast<MacroExpansionDecl>(D));
```

Comment:
In the future it might make sense to support functions inside enums etc, or computed properties etc.  But since `SelectiveTransform` is off by default we can expand the functionality here in later PRs without affecting the default case.

Reply:
Thinking about this some more, we may not need this outer decl check.  I'm going to take a look at making this logic more generalized.